lib/view_component/base.rb

```diff
@@ -279,6 +279,22 @@ def content_evaluated?
     #
     mattr_accessor :generate_stimulus_controller, instance_writer: false, default: false
 
+    # Always generate translation file alongside the component:
+    #
+    #     config.view_component.generate_locale = true
+    #
+    # Defaults to `false`.
+    #
+    mattr_accessor :generate_locale, instance_writer: false, default: false
+
+    # Always generate one translations file per available locale:
+    #
+    #     config.view_component.generate_splitted_locale_files = true
+    #
+    # Defaults to `false`. One file will be generated for each configured `I18n.available_locales`.
+    #
+    mattr_accessor :generate_splitted_locale_files, instance_writer: false, default: false
```

Comment:
What about renaming the option to use the word `distinct`, as you did in the generators docs? It feels clearer to me.

```suggestion
    mattr_accessor :generate_distinct_locale_files, instance_writer: false, default: false
```

Reply:
Seems right to me. Thanks for your hawk-coherence-eye! ;)